spring-boot-project/spring-boot-docker-compose/src/main/java/org/springframework/boot/docker/compose/service/connection/postgres/PostgresJdbcUrl.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2012-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.docker.compose.service.connection.postgres;
+
+import java.util.StringTokenizer;
+
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+/**
+ * Utility class to customize Postgres JDBC URL.
+ *
+ * @author Dmytro Nosan
+ */
+class PostgresJdbcUrl {
```

Comment:
We discussed this today and it feels bit overly complicated to us. We think the same could be achieved with something similar to the code that [disables encryption for SQL Server](https://github.com/spring-projects/spring-boot/blob/69162f4f9fdf373c418ebb89165f19b48ef5c4cf/spring-boot-project/spring-boot-docker-compose/src/main/java/org/springframework/boot/docker/compose/service/connection/sqlserver/SqlServerJdbcDockerComposeConnectionDetailsFactory.java#L62-L72). One addition that's probably needed here is to encode the application name as it's user-provided input and may contain characters that are restricted in a query string.

Reply:
Fixed